### ISSUE TITLE
Easier toggling ‘display/hide events by clicking on colored dot

### ISSUE THREAD

**Issue by JustACasual** [Etar-Group/Etar-Calendar#1172]
Feature Request: 
Currently, if you tap anywhere in a calendar line unser 'settings', you go to the detailed settings of this calendar.
This is also necessary for hiding/displaying a certain calendar. 
Would it be possible to make a tap to the colored dot / circle toggle the hide / display switch for this calendar?
The dot / circle indicator already serves as display for this switch (filled / unfilled), so it would feel quite ergonomic.
![Screenshot_20220531-213950_1_1.png](https://user-images.githubusercontent.com/55889353/171272626-7a405d1f-6e44-41a1-9d6e-64c725eb3b9c.png)

**Comment by JustACasual** [comment]
I would be willing to contribute on this, as I really think it is a useful feature with no disadvantages.
Could someone give me a tip on where to start digging?

Thank you :)

**Comment by JustACasual** [comment]
Is somebody reading this?

**Comment by Cwpute** [comment]
> Is somebody reading this?

I am! now if you're talking about knowledgeable people who already dipped their hands into the code, the i'm not. Your best bet might be to "just do it" on your own branch and ask for a merge later on.

A similar suggestion was already made 2 years ago and @newhinton even wrote and submitted [some modifications](https://github.com/Etar-Group/Etar-Calendar/pull/734) which were ultimately not accepted:
* https://github.com/Etar-Group/Etar-Calendar/issues/730

Here are two other related issues i found asking for the same kind of feature:
* https://github.com/Etar-Group/Etar-Calendar/issues/939
* https://github.com/Etar-Group/Etar-Calendar/issues/515

**Comment by Cwpute** [comment]
Currently it is quite tedious to select which calendars you want to show and which ones you don't want to. You have to open the side menu, go to Parameters, and for each calendar you want to change the visibility of, open its options and elect the visibility.
This is slow and unintuitive.

I personally switch between several calendars constantly as to not show everything at once, which would be horribly unreadable. Doing that directly from the main view would greatly help, and be quite simple too.

I've provided a preview of how it could work. A simple switch button next to the calendar's name would suffice. Here is the mock-up: 
![preview agenda view](https://user-images.githubusercontent.com/80604512/221533752-4998bf6b-467f-4ddb-b741-319a44b69264.jpg)

**Comment by jdaviescoates** [comment]
I think the best bet would be to combine both of your proposals, i.e.

When clicking the hamburger menu it should display all the calendars and their dots in the space underneath About Calendar and then clicking on those dots would show/ hide each calendar.

![Screenshot_20230227-215844_Calendar](https://user-images.githubusercontent.com/2113743/221695706-994bde48-5765-4d79-992a-0775c4c46311.png)

As a bonus there could also be a UI for saying "display only this calendar", like Google Calendar has.

**Comment by Cwpute** [comment]
Indeed, very good compromise here ! would both allow for quick visibility toggles and provide all the info you want about the calendars

**Comment by LnLcFlx** [comment]
I agree! Also, I'd like to add the suggestion to make this "visible calendars" selection *per view* (saved separately for Day, Week, Month, Agenda view).

For example, I do not wish to see my timetable calendar of my lectures in the Month view for it would clutter my entire view and hide the important events.
I do, however, want to see my lectures in the Week view.

For me this would be the most useful change to the entire app.
